routes/merged/project/location/private_subnet.rb

```diff
@@ -0,0 +1,97 @@
+# frozen_string_literal: true
+
+class Clover
+  branch = lambda do |r|
+    r.get api? do
+      private_subnet_list
+    end
+
+    r.on NAME_OR_UBID do |ps_name, ps_id|
+      if ps_name
+        r.post true do
+          private_subnet_post(ps_name)
+        end
+
+        filter = {Sequel[:private_subnet][:name] => ps_name}
+      else
+        filter = {Sequel[:private_subnet][:id] => UBID.to_uuid(ps_id)}
+      end
+
+      filter[:location] = @location
+      ps = @project.private_subnets_dataset.first(filter)
+
+      unless ps
+        response.status = request.delete? ? 204 : 404
+        next
+      end
+
+      if web?
+        r.post "connect" do
+          Authorization.authorize(current_account.id, "PrivateSubnet:connect", ps.id)
+          subnet = PrivateSubnet.from_ubid(r.params["connected-subnet-ubid"])
+          Authorization.authorize(current_account.id, "PrivateSubnet:connect", subnet.id)
+          ps.connect_subnet(subnet)
+          flash["notice"] = "#{subnet.name} will be connected in a few seconds"
+          r.redirect "#{@project.path}#{ps.path}"
+        end
+
+        r.post "disconnect", String do |disconnecting_ps_ubid|
+          Authorization.authorize(current_account.id, "PrivateSubnet:disconnect", ps.id)
+          subnet = PrivateSubnet.from_ubid(disconnecting_ps_ubid)
+          Authorization.authorize(current_account.id, "PrivateSubnet:disconnect", subnet.id)
+          ps.disconnect_subnet(subnet)
+          flash["notice"] = "#{subnet.name} will be disconnected in a few seconds"
+          ""
+        end
+      end
+
+      request.get true do
+        Authorization.authorize(current_account.id, "PrivateSubnet:view", ps.id)
+        @ps = Serializers::PrivateSubnet.serialize(ps)
+        if api?
+          @ps
+        else
+          @nics = Serializers::Nic.serialize(ps.nics)
+          @connected_subnets = Serializers::PrivateSubnet.serialize(ps.connected_subnets)
+          connectable_subnets = ps.projects.first.private_subnets.select do |ps1|
+            ps1_id = ps1.id
+            ps1_id != ps.id && !ps.connected_subnets.find { |cs| cs.id == ps1_id }
```

Comment:
Why don't we call `ps1.id` directly?
```suggestion
            ps1.id != ps.id && !ps.connected_subnets.find { |cs| cs.id == ps1.id }
```

Reply:
It's a micro-optimization.  Using a local variable saves 1 method call per `find` iteration.

Reply:
Also, note that this is inside another `select` loop, so the number of method calls saved is `select` iterations * `find` iterations.